nexus/src/db/datastore.rs

```diff
@@ -2679,6 +2679,70 @@ impl DataStore {
             }),
         }
     }
+
+    // SSH public keys
+
+    pub async fn ssh_keys_list(
+        &self,
+        silo_user_id: Uuid,
+        page_params: &DataPageParams<'_, Name>,
+    ) -> ListResultVec<SshKey> {
+        use db::schema::ssh_key::dsl;
+
+        paginated(dsl::ssh_key, dsl::name, page_params)
+            .filter(dsl::silo_user_id.eq(silo_user_id))
+            .filter(dsl::time_deleted.is_null())
+            .select(SshKey::as_select())
+            .load_async(self.pool())
+            .await
+            .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
+    }
+
+    /// Create a new SSH public key for a user.
+    pub async fn ssh_key_create(
+        &self,
+        ssh_key: SshKey,
+    ) -> CreateResult<SshKey> {
+        use db::schema::ssh_key::dsl;
+
+        diesel::insert_into(dsl::ssh_key)
+            .values(ssh_key)
+            .returning(SshKey::as_returning())
+            .get_result_async(self.pool())
+            .await
+            .map_err(|e| {
+                Error::internal_error(&format!(
+                    "error creating SSH key: {:?}",
+                    e
+                ))
+            })
+    }
+
+    /// Delete an existing SSH public key.
+    pub async fn ssh_key_delete(
+        &self,
+        opctx: &OpContext,
+        authz_ssh_key: &authz::SshKey,
+    ) -> DeleteResult {
+        opctx.authorize(authz::Action::Delete, authz_ssh_key).await?;
+
+        use db::schema::ssh_key::dsl;
+
+        let now = Utc::now();
+        diesel::update(dsl::ssh_key)
+            .filter(dsl::id.eq(authz_ssh_key.id()))
+            .filter(dsl::time_deleted.is_null())
```

Comment:
I think you want to use `.check_if_exists::<SshKey>(authz_ssh_key.id())` here.  I see we're doing this in `volume_delete()`, but I see we're _not_ doing it in some other places that I think we should be (like `instance_delete_network_interface` or `project_delete_vpc`).  Dang.

Reply:
Fixed in 95114e05.